app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx

```diff
@@ -42,7 +44,14 @@ export default class SampleViewV2 extends React.Component {
   constructor(props) {
     super(props);
 
-    let localState = this.loadState(localStorage, "SampleViewOptions");
+    this.urlParser = new UrlQueryParser({
+      pipelineVersion: "string",
+      selectedOptions: "object",
+    });
+    this.urlOptions = this.urlParser.parse(location.search);
+
+    const urlState = pick("selectedOptions", this.urlOptions);
+    const localState = this.loadState(localStorage, "SampleViewOptions");
```

Comment:
I do not think this single line justifies the an extra API just for this... But since this is the second time we do it, I'm thinking of ways of generalizing the whole pushing to browsing and to state in the future.